### ISSUE TITLE
[Request] Generic binary releases

### ISSUE THREAD

**Issue by RivenSkaye** [elizagamedev/vobsubocr#17]
Would it be possible to have tags be accompanied by generic binary builds to more easily use this program for people who don't run Rust toolchains (yet)?
A friend of mine wants to use this, but doesn't have a clue how to handle fairly obvious problems like missing vcpkg and libclang.dll, a binary release would make it a lot more accessible to less tech savvy people

**Comment by gwen-lg** [comment]
For which platform you would like binaries ?

**Comment by RivenSkaye** [comment]
x86_64 would be the most common target considering most modern hardware right? That seems like a safe bet anyone can run

**Comment by gwen-lg** [comment]
And the Os ? I probably can provide a Linux binary relatively easily, but for Windows, it's probably less simple for me.

**Comment by RivenSkaye** [comment]
Windows was the idea, considering target audience of less tech savvy people. For Linux releases I have no idea what the most convenient package format would be tbh. I suppose a plain ELF binary would be easiest for people to wrap into a mininal package themselves, or to dump in whatever utility dir lives on their PATH. Building it on whatever Ubuntu stable ships would allow for the most widespread compatibility thanks to clean glibc versioning, which would make it relatively easy to leverage thw GH workflow runners to build them if you don't want to manually build them every time

**Comment by gwen-lg** [comment]
I have take a look, but because of the dependencies, it's not trivial.
What do you think about a "online" conversion ?
Sort of "VobsubOcr as a service"

As alternative, there is application like Subtitle Composer, even if there don't use OCR.

**Comment by languagemaniac** [comment]
Hi, i'm quite a noob and just can't build this. Could you please provide a binary for 64 bit linux? or arm64 for raspberry pi running raspberry pi os (debian bullseye)

**Comment by gwen-lg** [comment]
@languagemaniac can you test this https://github.com/gwen-lg/vobsubocr/releases/tag/v2023-12-13 please ?
For a amr64 version I need more work to do cross-compilation.

PS: If it's work and you can, a tips on https://liberapay.com/gwenlg/ to thank me are welcome ;)

**Comment by languagemaniac** [comment]
> @languagemaniac can you test this https://github.com/gwen-lg/vobsubocr/releases/tag/v2023-12-13 please ?
> For a amr64 version I need more work to do cross-compilation.
> 
> PS: If it's work and you can, a tips on https://liberapay.com/gwenlg/ to thank me are welcome ;) 
> 

Thanks! Unfortunately I'm a student, but I do plan to support many Foss projects like yours in a hopefully near future.

I'll test and update this tomorrow

**Comment by kkm** [comment]
![image](https://github.com/elizagamedev/vobsubocr/assets/134290/6537ca86-519d-4571-a543-b8e185f6c862)
@gwen-lg I think, it needs library.

**Comment by RivenSkaye** [comment]
@kkm try searching your package repositories for `libtessaract5` or similar names; even Debian sid is already on 5.3.
Worst comes to worst, you can always grab a copy [of the 5.3.2 source code](https://github.com/tesseract-ocr/tesseract/releases/tag/5.3.2) and build it yourself

**Comment by gwen-lg** [comment]
Yes I forgot to mention, vobsubocr depends on tesseract and leptonica, so you need to have it installed.
But in addition, it should be binary compatible, which is not necessarily the case.

@RivenSkaye I think it should be more easy to directly build vobsubocr, it's relatively easy with cargo.

**Comment by kkm** [comment]
> @kkm try searching your package repositories for `libtessaract5` or similar names; even Debian sid is already on 5.3. Worst comes to worst, you can always grab a copy [of the 5.3.2 source code](https://github.com/tesseract-ocr/tesseract/releases/tag/5.3.2) and build it yourself

@RivenSkaye 
![image](https://github.com/elizagamedev/vobsubocr/assets/134290/179ad2e3-f79e-4461-ba19-3661a8739244)
I did install with tesseract 5.3.2 on Ubuntu 20.04, but it doesnt work with vobsubocr-x86_64, maybe different libtessaract5 depends?

**Comment by RivenSkaye** [comment]
I specifically linked 5.3.2 due to the error message you got. Can you run `sudo updatedb && locate libtesseract`? (If you don't have a `locate` installed, `sudo apt install plocate`)

**Comment by gwen-lg** [comment]
As alternative to build yourself a version, what do you think of an online service like : https://sub-serv.gwenlg.fr

Note: It's a preview webapp, the the service can be down sometimes. And I opened to comment, problem report but be nice with it and me.
Thank you.
